src/bin/wrapped-wit.js

```diff
@@ -0,0 +1,1029 @@
+        proceed = true
+        break;
+
+      case 1n: // Awaiting
+        const user = await prompt([{
+          message: `The specified VTT hash is currently being verified. Shall we retry, anyways ?`,
+          name: "continue",
+          type: "confirm",
+          default: false,
+        }])
+        proceed = user.continue;
+        break;
+
+      default:
+        proceed = false
+    }
+    
+    if (proceed && vtt.finalized !== 1) {
+      console.info(helpers.colors.mred(`\n => Sorry, wait a few minutes until the transaction gets finalized on Witnet,`))
+      console.info(helpers.colors.mred(`      before querying on-chain verification.\n`))
+      proceed = false
+    }
+
+    if (proceed) {
+      if (!gasPrice) gasPrice = (await provider.getFeeData()).gasPrice;
+      let fee = await contract.witOracleEstimateWrappingFee(gasPrice)
+      const gas = await contract.wrap.estimateGas(`0x${vttHash}`, { value: fee, gasPrice })
+      let cost = fee + gasPrice * gas
+      const user = await prompt([{
+        message: `Verification is expected to cost ${ethers.formatEther(cost)} ETH. Shall we proceed ?`,
+        name: "continue",
+        type: "confirm",
+        default: true,
+      }])
+      if (user.continue) {
+        // connect contract to the eth/rpc provider's default signer
+        contract = contract.connect(await provider.getSigner())
+
+        console.info(helpers.colors.lwhite(`> Wrapping ${ethers.formatUnits(vtt.value, 9)} WIT ...`))
+        console.info(`  - WITNET sender:  ${vtt.sender}`)
+        console.info(`  - EVM recipient:  ${ethers.getAddress(`0x${vtt.metadata}`)}`)
+        console.info(`  - EVM signer:     ${(await provider.getSigner()).address}`)
+        console.info(`  - Wit/Oracle fee: ${ethers.formatEther(fee)} ETH`)
+        
+        await contract
+          .wrap
+          .send(`0x${vttHash}`, { value: fee, gasPrice })
+          .then(async (tx) => {
+            console.info(`  - EVM tx hash:    ${tx.hash}`)
+            return await helpers.prompter(tx.wait(confirmations || 1))
+          })
+          .then(receipt => {
+            console.info(`  - EVM tx cost:    ${ethers.formatEther(fee + receipt.gasPrice * receipt.gasUsed)} ETH`)
+            console.info(`  - Gas price:      ${helpers.commas(receipt.gasPrice)}`)
+            console.info(`  - Gas used:       ${helpers.commas(receipt.gasUsed)}`)
+            console.info(`  - Block number:   ${helpers.commas(receipt.blockNumber)}`)
+            return receipt
+          });
+        // settle --from filter
+        from = vtt.sender
+        // delete --into filter
+        delete into
+      }
+    }
+  }
+
+  const fromBlock = since ? BigInt(since) : undefined
+  let events = (fromBlock !== undefined && !value
+    ? (await contract.queryFilter("Wrapped", fromBlock)).slice(flags?.offset || 0, flags?.limit || 64)
+    : (await contract.queryFilter("Wrapped")).reverse().slice(0, flags?.limit || 64)
+  );
```

Comment:
```suggestion
const fromBlock = since ? BigInt(since) : undefined
const limit     = flags?.limit ?? 64

let events = await contract.queryFilter('Wrapped', fromBlock)

if (fromBlock !== undefined && !value) {
  events = events.slice(flags?.offset ?? 0, limit)
} else {
  events = events.reverse().slice(0, limit)
```

Reply:
Sorry, that'd be changing the intended logic.